### ISSUE TITLE
[v6.2.35] Allow RestTotalHitsAsInt on Scroll API

### ISSUE THREAD

**Issue by athap** [olivere/elastic#1436]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
[x] elastic.v6 (for Elasticsearch 6.x)
[ ] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior
svc := client.Scroll(testIndexName).Size(1).RestTotalHitsAsInt(true) should work 

### Please describe the actual behavior
Current Scroll service fails in ES 7 with the following error 

```
cannot unmarshal object into Go struct field SearchHits.hits.total of type int64
```

**Comment by athap** [comment]
Willing to work on a PR if this something that can be done. Thank you

**Comment by olivere** [comment]
Fixed in bfa6e786e7d3613a20c55c2ebeb099e8b04be8b0.
